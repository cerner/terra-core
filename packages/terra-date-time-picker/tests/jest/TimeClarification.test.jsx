import React from 'react';
import { IntlProvider } from 'react-intl';
import TimeClarification from '../../lib/_TimeClarification';
import messages from '../../translations/en-US.json';

const locale = 'en-US';

it('should render a default date time picker', () => {
  const datePicker = shallow((
    <IntlProvider locale={locale} messages={messages}>
      <TimeClarification
        isOpen
        isOffsetButtonHidden
        onDaylightSavingButtonClick={jest.fn()}
        onStandardTimeButtonClick={jest.fn()}
        onOffsetButtonClick={jest.fn()}
        onRequestClose={jest.fn()}
        releaseFocus={jest.fn()}
        requestFocus={jest.fn()}
      />
    </IntlProvider>));
  expect(datePicker).toMatchSnapshot();
});

it('should render a disabled time clarification', () => {
  const datePicker = shallow((
    <IntlProvider locale={locale} messages={messages}>
      <TimeClarification
        isOpen
        isOffsetButtonHidden
        onDaylightSavingButtonClick={jest.fn()}
        onStandardTimeButtonClick={jest.fn()}
        onOffsetButtonClick={jest.fn()}
        onRequestClose={jest.fn()}
        releaseFocus={jest.fn()}
        requestFocus={jest.fn()}
        disabled
      />
    </IntlProvider>));
  expect(datePicker).toMatchSnapshot();
});
