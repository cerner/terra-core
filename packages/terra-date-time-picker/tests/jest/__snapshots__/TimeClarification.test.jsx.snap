// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a default date time picker 1`] = `
<TimeClarification
  disabled={false}
  isOffsetButtonHidden={true}
  isOpen={true}
  onDaylightSavingButtonClick={[MockFunction]}
  onOffsetButtonClick={[MockFunction]}
  onRequestClose={[MockFunction]}
  onStandardTimeButtonClick={[MockFunction]}
  releaseFocus={[MockFunction]}
  requestFocus={[MockFunction]}
/>
`;

exports[`should render a disabled time clarification 1`] = `
<TimeClarification
  disabled={true}
  isOffsetButtonHidden={true}
  isOpen={true}
  onDaylightSavingButtonClick={[MockFunction]}
  onOffsetButtonClick={[MockFunction]}
  onRequestClose={[MockFunction]}
  onStandardTimeButtonClick={[MockFunction]}
  releaseFocus={[MockFunction]}
  requestFocus={[MockFunction]}
/>
`;
